python/cris_import.py

```diff
@@ -0,0 +1,257 @@
+  insert_db_people_cris(
+    objects: $objects, 
+    on_conflict: {
+        constraint: unique_people_cris,
+        update_columns: [$updateColumns]
+    }) {
+    affected_rows
+  }
+}"""
+
+mutations = {
+    "crashes": CRASH_UPSERT_MUTATION,
+    "units": UNIT_UPSERT_MUTATION,
+    "persons": PERSON_UPSERT_MUTATION,
+}
+
+
+def make_upsert_mutation(table_name):
+    """Sets the column names in the upsert on conflict array"""
+    upsert_mutation = mutations[table_name]
+    updateColumns = [
+        col["name"]
+        for col in COLUMNS
+        if col["table_name"] == table_name
+        and col["source"] == "cris"
+        and not col["is_primary_key"]
+    ]
+    return upsert_mutation.replace("$updateColumns", ", ".join(updateColumns))
+
+
+def make_hasura_request(*, query, endpoint, variables=None):
+    payload = {"query": query, "variables": variables}
+    res = requests.post(endpoint, json=payload)
+    res.raise_for_status()
+    data = res.json()
+    try:
+        return data["data"]
+    except KeyError:
+        raise ValueError(data)
+
+
+def get_file_meta(filename):
+    """
+    Returns:
+        schema_year, extract_id, table_name
+    """
+    filename_parts = filename.split("_")
+    return filename_parts[1], filename_parts[2], filename_parts[3]
+
+
+def get_files_todo():
+    files_todo = []
+    for filename in os.listdir(FILE_DIR):
+        if not filename.endswith(".csv"):
+            continue
+        schema_year, extract_id, table_name = get_file_meta(filename)
+
+        # ignore all tables except these
+        if table_name not in ("crash", "unit", "person", "primaryperson"):
+            continue
+
+        file = {
+            "schema_year": schema_year,
+            "table_name": table_name,
+            "path": os.path.join(FILE_DIR, filename),
+            "extract_id": extract_id,
+        }
+        files_todo.append(file)
+    return files_todo
+
+
+def load_csv(filename):
+    with open(filename, "r") as fin:
+        reader = csv.DictReader(fin)
+        return [row for row in reader]
+
+
+def lower_case_keys(list_of_dicts):
+    return [{key.lower(): value for key, value in row.items()} for row in list_of_dicts]
+
+
+def remove_unsupported_columns(rows, table_name):
+    col_names_to_keep = [
+        col["name"]
+        for col in COLUMNS
+        if col["table_name"] == table_name and col["source"] == "cris"
+    ]
+    return [
+        {key: val for key, val in row.items() if key in col_names_to_keep}
+        for row in rows
+    ]
+
+
+def handle_empty_strings(rows):
+    for row in rows:
+        for key, val in row.items():
+            if val == "":
+                row[key] = None
+    return rows
+
+
+def handle_crash_date_time(rows):
+    """Combine crash_date and crash_time and format as ISO string.
+
+    The same ISO timestamp is stored as `crash_date` and `crash_time`
+    """
+    tzinfo = ZoneInfo("America/Chicago")
```

Comment:
i just wanted to mention that this is my first time using Python's new timezone support that came out in v3.9. it replaces `pytz` and it is great! cc @Charlie-Henry